server/platform/shared/filestore/s3store.go

```diff
@@ -571,20 +609,55 @@ func (b *S3FileBackend) RemoveDirectory(path string) error {
 	return nil
 }
 
-func (b *S3FileBackend) prefixedPath(s string) string {
+// prefixedPathFast is a variation of prefixedPath
+// where we don't check for the file path. This is for cases
+// where we know the file won't exist - like while writing a new file.
+func (b *S3FileBackend) prefixedPathFast(s string) string {
 	if b.isCloud {
-		// If the request is routed via bifrost, then we need to encode the path
-		// to avoid signature validation errors.
-		// This happens because in bifrost, we are signing the URL outside the SDK
-		// and therefore the signature sent from the bifrost client
-		// will contain the encoded path, whereas the original path is sent
-		// un-encoded.
-		// More info at: https://github.com/aws/aws-sdk-go/blob/a57c4d92784a43b716645a57b6fa5fb94fb6e419/aws/signer/v4/v4.go#L8
 		s = s3utils.EncodePath(s)
 	}
 	return filepath.Join(b.pathPrefix, s)
 }
 
+func (b *S3FileBackend) lookupOriginalPath(s string) (bool, error) {
+	exists, err := b._fileExists(filepath.Join(b.pathPrefix, s))
+	if err != nil {
+		var s3Err s3.ErrorResponse
+		// Sometimes S3 will not allow to access other paths.
+		// In that case, we consider them as not exists.
+		if errors.As(err, &s3Err); s3Err.Code == "AccessDenied" {
+			return false, nil
+		}
```

Comment:
Could this error be a permission error? If so, we would silently ignoring an error. Maybe we should log it at least?

Reply:
Yeah, happens on 404 cases.